libs/dtm0/dtm0_lib.py

```diff
@@ -0,0 +1,159 @@
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+"""
+Library Methods for DTM0 delivery testing
+"""
+import logging
+import random
+import time
+
+from config import S3_CFG
+from libs.s3 import ACCESS_KEY, SECRET_KEY
+from scripts.s3_bench import s3bench
+
+
+class DTM0TestLib:
+
+    def __init__(cls, access_key=ACCESS_KEY, secret_key=SECRET_KEY):
+        """
+        Init method
+        :param access_key: Access key for S3bench operations.
+        :param secret_key: Secret key for S3bench operations.
+        """
+        cls.log = logging.getLogger(__name__)
+        cls.access_key = access_key
+        cls.secret_key = secret_key
+
+    def perform_write_op(self, bucket_prefix, object_name, clients, samples, size,
+                         log_file_prefix, queue, loop=1):
+        """
+        Perform Write operations
+        :param bucket_prefix: Bucket name
+        :param object_name: Object name
+        :param clients: No of Client session
+        :param samples: No of samples
+        :param size: Object size
+        :param log_file_prefix: Log file prefix
+        :param queue: Multiprocessing Queue to be used for returning values (Boolean,dict)
+        :param loop: Loop count for writes
+        """
+        results = list()
+        workload = list()
+        log_path = None
+        for i in range(loop):
+            self.log.info("Perform Write Operations : ")
+            bucket_name = bucket_prefix + str(int(time.time()))
+            resp = s3bench.s3bench(self.access_key,
+                                   self.secret_key, bucket=bucket_name,
+                                   num_clients=clients, num_sample=samples,
+                                   obj_name_pref=object_name, obj_size=size,
+                                   skip_cleanup=True, duration=None,
+                                   log_file_prefix=str(log_file_prefix).upper(),
+                                   end_point=S3_CFG["s3_url"],
+                                   validate_certs=S3_CFG["validate_certs"])
+            self.log.info("Workload: %s objects of %s with %s parallel clients.",
+                          samples, size, clients)
+            self.log.info("Log Path %s", resp[1])
+            log_path = resp[1]
+            if s3bench.check_log_file_error(resp[1]):
+                results.append(False)
+                break
+            else:
+                results.append(True)
+                workload.append({'bucket': bucket_name, 'obj_name_pref': object_name,
+                                 'num_clients': clients, 'obj_size': size,
+                                 'num_sample': samples})
+        if all(results):
+            queue.put([True, workload])
+        else:
+            queue.put([False, f"S3bench workload for failed."
+                             f" Please read log file {log_path}"])
+
+    def perform_read_op(self, workload_info: list, queue, skipread: bool = True,
```

Comment:
Preferable to use ops as this function is not performing only single operation.
```suggestion
    def perform_read_ops(self, workload_info: list, queue, skipread: bool = True,
```